_extensions/acronyms/acronyms_helpers.lua

```diff
@@ -104,5 +104,272 @@ function Helpers.str_to_boolean(value)
+function Helpers.is_inline_array(tbl)
+    if type(tbl) ~= "table" then return false end
+    for i, v in ipairs(tbl) do
+        if type(v) ~= "table" or v.t == nil then return false end
+    end
+    return #tbl > 0
+end
+
+
+-- Transform inlines or strings according to case_kind while preserving inline structure.
+function Helpers.transform_case(value, case_kind)
+    if case_kind == nil then return value end
+
+    -- String values
+    if type(value) == "string" then
+        if case_kind == "upper" then return value:upper()
+        elseif case_kind == "lower" then return value:lower()
+        elseif case_kind == "sentence" then return Helpers.capitalize_first(value)
+        else return value end
+    end
+
+    -- If it's not a plain inline array, check for Pandoc Inlines/List
+    if not Helpers.is_inline_array(value) then
+        if pandoc.utils and pandoc.utils.type then
+            local t = pandoc.utils.type(value)
+            if t ~= "Inlines" and t ~= "List" then
+                return value
+            end
+        else
+            return value
+        end
+    end
+
+    local done_first = false
+    local simple_containers = {
+        Emph=true, Strong=true, Span=true, Strikeout=true,
+        SmallCaps=true, Superscript=true, Subscript=true, Underline=true
+    }
+
+    local function transform_inlines(src)
+        local dest = {}
+        for _, il in ipairs(src) do
+            if il.t == "Str" and (il.text or il.c) then
+                local txt = il.text or il.c
+                if case_kind == "upper" then
+                    txt = txt:upper()
+                elseif case_kind == "lower" then
+                    txt = txt:lower()
+                elseif case_kind == "sentence" and not done_first then
+                    local i = txt:find("%a")
+                    if i then
+                        txt = txt:sub(1,i-1)..txt:sub(i,i):upper()..txt:sub(i+1)
+                        -- set done_first to ensure only one capitalization in sentence case
+                        done_first = true
+                    end
+                end
+                dest[#dest+1] = pandoc.Str(txt)
+            elseif simple_containers[il.t] and type(il.c) == "table" then
+                local inner = transform_inlines(il.c)
+                if il.t == "Span" then
+                    dest[#dest+1] = pandoc.Span(inner, il.attr)
+                else
+                    local ctor = pandoc[il.t]
+                    if ctor then
+                        dest[#dest+1] = ctor(inner)
+                    else
+                        local copy = {}
+                        for k, v in pairs(il) do copy[k] = v end
+                        copy.c = inner
+                        dest[#dest+1] = copy
+                    end
+                end
+            else
+                dest[#dest+1] = il
+            end
+        end
+        return dest
+    end
+
+    return transform_inlines(value)
+end
```

Comment:
Is it necessary to put this function here in Helpers? My idea for Helpers was to put here functions that can be used in several places. I think that `transform_case` is only used when rendering an acronym, so maybe we could place this in `acronyms_styles`? (or perhaps `acronyms_pandoc`?)